components/popconfirm/popconfirm.ts

```diff
@@ -289,16 +286,18 @@ export class NzPopconfirmComponent extends NzToolTipComponent implements OnDestr
 
   onConfirm(): void {
     if (this.nzBeforeConfirm) {
-      const observable = wrapIntoObservable(this.nzBeforeConfirm()).pipe(first());
       this.confirmLoading = true;
-      observable
+      this.cdr.markForCheck();
+
+      wrapIntoObservable(this.nzBeforeConfirm())
         .pipe(
+          first(),
           finalize(() => {
             this.confirmLoading = false;
             this.cdr.markForCheck();
           }),
           takeUntil(this.nzVisibleChange),
-          takeUntil(this.destroy$)
+          takeUntilDestroyed(this.destroyRef)
```

Comment:
I thinks it's not needed to have a takeUntil operator here, cause we use first RxJs Operator which complete the subscription after send the first